src/osiris_log.erl

```diff
@@ -376,7 +380,8 @@
          transport :: tcp | ssl}).
 -record(write,
         {type = writer :: writer | acceptor,
-         segment_size = 0 :: non_neg_integer(),
+         segment_size_bytes = 0 :: non_neg_integer(),
```

Comment:
as these are always written together you could consider making them a tuple instead of two separate fields. it is good to keep records that are updated often as small as possible as every update is a full tuple copy.